sympy/physics/control/lti.py

```diff
@@ -1144,3 +1198,164 @@ def _eval_rewrite_as_TransferFunction(self, num, den, **kwargs):
 
     def __neg__(self):
         return Feedback(-self.num, self.den)
+
+
+class TransferFunctionMatrix(Basic):
+
+    def __new__(cls, arg):
+        if not (isinstance(arg, list) and (all(isinstance(i, list) for i in arg) or
+            all(isinstance(i, TransferFunction) for i in arg))):
+            raise TypeError("Unsupported type for argument of TransferFunctionMatrix.")
+
+        obj = super(TransferFunctionMatrix, cls).__new__(cls, arg)
```

Comment:
This can make use of an extra parameter to store shape attributes

Reply:
I think using `Basic.__new__(cls, arg, shape, var)` would be unnecessary because `shape` and `var` attributes can be derived from `arg`.

Reply:
Unfortunately, you were storing _inputs, _outputs as mutable parametets and it is worse than storing them in slots if it is not necessary

Reply:
OK, I will apply this suggestion. Also, note that the `str` representation of `Series` and `Parallel` will get way too longer. Just saying... 
Anyways, It's better for the user to use pretty-printing because normal printing is already too long and unreadable for every object.